pkg/kubeadm/cmd/init.go

```diff
@@ -67,15 +67,31 @@ func NewCmdInit(out io.Writer, s *kubeadmapi.KubeadmConfig) *cobra.Command {
 
 	cmd.PersistentFlags().IPNetVar(
 		&s.InitFlags.Services.CIDR, "service-cidr", *kubeadmapi.DefaultServicesCIDR,
-		`(optional) use alterantive range of IP address for service VIPs, e.g. "10.16.0.0/12"`,
+		`(optional) use alternative range of IP address for service VIPs, e.g. "10.16.0.0/12"`,
 	)
 	cmd.PersistentFlags().StringVar(
 		&s.InitFlags.Services.DNSDomain, "service-dns-domain", kubeadmapi.DefaultServiceDNSDomain,
-		`(optional) use alterantive domain name for services, e.g. "myorg.internal"`,
+		`(optional) use alternative domain name for services, e.g. "myorg.internal"`,
 	)
 	cmd.PersistentFlags().StringVar(
 		&s.InitFlags.CloudProvider, "cloud-provider", "",
-		`(optional) enable cloud proiver features (external load-balancers, storage, etc)`,
+		`(optional) enable cloud provider features (external load-balancers, storage, etc)`,
+	)
+	cmd.PersistentFlags().StringSliceVar(
```

Comment:
Could you take a look and see what it would that to make these flags no appear in `--help` and only appear in `--advanced-help`?